teps/0115-tekton-catalog-git-based-versioning.md

```diff
@@ -0,0 +1,680 @@
+    /README.md
+    /openshift-client-python.yaml
+    /Dockerfile
+  /openshift-client
+    /README.md
+    /openshift-client.yaml
+    /tests
+    /samples
+    /OWNERS
+  /openshift-install
+    /README.md
+    /openshift-client.yaml
+    /tests
+  /openshift-uninstall
+    /README.md
+    /openshift-client.yaml
+    /tests
+```
+
+The Openshift team can share this Catalog with the `Tasks` in the Hub without adding it to a centralized Catalog. 
+They can then version the Catalog and its resources using release tags, which would be picked up by the Tekton Hub.
+
+Note that the version tags used in this Catalog applies to all the resources. For example, if the Openshift team
+updates the "openshift-client" `Task` and makes a new release tagged "v0.5" then there will be a version "0.5" for
+all the resources in the Catalog. If the Openshift team would prefer to decouple the versioning of a specific resource
+from the rest, they can isolate the resources into its own Catalog where it will have its own versioning.
+
+#### Migration
+
+The Hub should support both the directory-based and git-based versioning to provide backwards compatibility. The 
+[config.yaml][cy] will take a `versioning` field under `catalogs`, which will default to `"directory"` and users can
+set it to `"git"`. Taking the examples above, the entries in [config.yaml][cy] could be:
+
+```yaml
+catalogs:
+
+- name: tekton
+  org: tektoncd
+  type: community
+  provider: github
+  url: https://github.com/tektoncd/catalog
+
+- name: buildpacks
+  org: buildpacks
+  type: community
+  provider: github
+  url: https://github.com/buildpacks/tekton-integration
+  versioning: git
+
+- name: slack-notify
+  org: ebay
+  type: community
+  provider: github
+  url: https://github.com/eBay/tekton-slack-notify
+  versioning: git
+
+- name: openshift
+  org: redhat
+  type: community
+  provider: github
+  url: https://github.com/openshift/tekton-catalog
+  versioning: git
+ ```
+
+The existing Tekton Catalog - `tektoncd/catalog` - will remain as it is using the directory-based contract. It will be 
+read-only and will encourage owners of the resources to move them to their own repositories in their own GitHub
+organizations. After 9 months, we will archive the `tektoncd/catalog` repository. At the same time, we will remove
+support for the directory-based versioning in the Tekton Hub.
+
+We will provide official Tekton resources through Catalogs in `tektoncd-catalog` - https://github.com/tektoncd-catalog.
+The official resources will be a small set of resources that are critical in running Tekton e.g. the "git-clone" `Task`.
+This GitHub organization will also contain a template Catalog that users can use on to get started easily. Owners of the
+official resources will be the Tekton Catalog Maintainers; we will update the contributor ladder for Catalog to reflect
+that contributions are distributed in multiple repositories. This will be scoped further in [TEP-0079][tep-0079].
```

Comment:
I think for this TEP to be marked as implementable we need a more detailed migration plan.
How will we setup and maintain the new GitHub org, who is responsible for moving resources into new repos in the new org, how do we setup and maintain CI and CD for all the new repos etc.

Reply:
The owners of the specific resources are responsible for migrating their own resources, while the tekton catalog maintainers are responsible for migrating the tekton official resources -- clarified this in the TEP

The testing infra will be provided in the Catalog template that'd make it easier for users to get started